app/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/claims/CheckClaimantDetailsController.scala

```diff
@@ -41,30 +56,97 @@ class CheckClaimantDetailsController @Inject() (
     extends FrontendController(cc)
     with WithAuthAndSessionDataAction
     with SessionUpdates
     with SessionDataExtractor
     with Logging {
 
-  implicit val dataExtractor: DraftC285Claim => Option[DetailsRegisteredWithCdsAnswer] =
-    _.detailsRegisteredWithCdsAnswer
+  implicit val dataExtractor: DraftC285Claim => Option[CheckClaimantDetailsAnswer] = _.checkClaimantDetailsAnswer
 
   def show(implicit journey: JourneyBindable): Action[AnyContent] =
     authenticatedActionWithSessionData.async { implicit request =>
-      withAnswersAndRoutes[DetailsRegisteredWithCdsAnswer] { (fillingOutClaim, _, router) =>
-        val namePhoneEmail       = extractContactsRegisteredWithCDSA(fillingOutClaim)
-        val establishmentAddress = extractEstablishmentAddress(fillingOutClaim)
-        Ok(claimantDetails(namePhoneEmail, establishmentAddress, router))
+      withAnswersAndRoutes[CheckClaimantDetailsAnswer] { (fillingOutClaim, answers, router) =>
+        val emptyForm  = checkClaimantDetailsAnswerForm
+        val filledForm = answers.fold(emptyForm)(emptyForm.fill(_))
+        Ok(renderTemplate(filledForm, fillingOutClaim, router))
       }
     }
 
+  def submit(implicit journey: JourneyBindable): Action[AnyContent] =
+    authenticatedActionWithSessionData.async { implicit request =>
+      withAnswersAndRoutes[CheckClaimantDetailsAnswer] { (fillingOutClaim, _, router) =>
+        checkClaimantDetailsAnswerForm
+          .bindFromRequest()
+          .fold(
+            formWithErrors => BadRequest(renderTemplate(formWithErrors, fillingOutClaim, router)),
+            formOk => {
+              val newDraftClaim  = fillingOutClaim.draftClaim.fold(_.copy(checkClaimantDetailsAnswer = Option(formOk)))
+              val updatedJourney = fillingOutClaim.copy(draftClaim = newDraftClaim)
+
+              EitherT(updateSession(sessionStore, request)(_.copy(journeyStatus = Some(updatedJourney))))
+                .leftMap(err => Error(s"Could not save Declarant Type: ${err.message}"))
+                .fold(
+                  e => {
+                    logger.warn("Submit Declarant Type error: ", e)
+                    errorHandler.errorResult()
+                  },
+                  _ => Redirect(router.nextPageForClaimantDetails())
+                )
+            }
+          )
+      }
+
+    }
+
+  def renderTemplate(
+    form: Form[CheckClaimantDetailsAnswer],
+    fillingOutClaim: FillingOutClaim,
+    router: ReimbursementRoutes
+  )(implicit
+    request: RequestWithSessionData[_],
+    messages: Messages,
+    viewConfig: ViewConfig
+  ): HtmlFormat.Appendable =
+    claimantDetails(
+      form,
+      extractContactsRegisteredWithCDSA(fillingOutClaim),
+      extractEstablishmentAddress(fillingOutClaim),
+      extractContactDetails(fillingOutClaim),
+      extractContactAddress(fillingOutClaim),
+      router
+    )
+
 }
 
 object CheckClaimantDetailsController {
   val languageKey: String = "claimant-details"
```

Comment:
'languageKey' implies something about changing/using languages.
messageKey is a better description.